src/effects/ObjectDetection.cpp

```diff
@@ -254,21 +343,46 @@ void ObjectDetection::SetJsonValue(const Json::Value root) {
 	// Set parent data
 	EffectBase::SetJsonValue(root);
 	// Set data from Json (if key is found)
-	if (!root["protobuf_data_path"].isNull()){
-		protobuf_data_path = (root["protobuf_data_path"].asString());
+	if (!root["protobuf_data_path"].isNull() && protobuf_data_path.size() <= 1){
+		protobuf_data_path = root["protobuf_data_path"].asString();
 
 		if(!LoadObjDetectdData(protobuf_data_path)){
 			std::cout<<"Invalid protobuf data path";
 			protobuf_data_path = "";
 		}
 	}
+
+    // Set the selected object index
+    if (!root["selected_object_index"].isNull())
+        selectedObjectIndex = root["selected_object_index"].asInt();
+
+    // Set the tracked object's ids
+    if (!root["objects_id"].isNull()){
+        for (auto const& trackedObject : trackedObjects){
+            Json::Value trackedObjectJSON;
+            trackedObjectJSON["box_id"] = root["objects_id"][trackedObject.first].asString();
+            trackedObject.second->SetJsonValue(trackedObjectJSON);
+        }
+    }
+
+    // Set the selected object's properties
+    auto selectedObject = trackedObjects.at(selectedObjectIndex);
+    if (selectedObject)
+        selectedObject->SetJsonValue(root);
 }
 
 // Get all properties for a specific frame
 std::string ObjectDetection::PropertiesJSON(int64_t requested_frame) const {
 
 	// Generate JSON properties list
 	Json::Value root;
+
+    // Add the selected object Json to root
```

Comment:
Nitpick: Indention, perhaps just scan all these files for indention issues and make them as close to the remaining source code as possible. Thx.